addons/sale/models/sale_order.py

```diff
@@ -367,7 +367,11 @@ def _compute_amount_undiscounted(self):
         for order in self:
             total = 0.0
             for line in order.order_line:
-                total += (line.price_subtotal * 100)/(100-line.discount) if line.discount != 100 else (line.price_unit * line.product_uom_qty)
+                undiscount = 100 - line._get_real_discount(line.discount)
```

Comment:
AFAICS, using your new util method in a non stored field will mean that the computed amount will vary depending on the pricelist configuration, whenever you display the record, potentially holding an incorrect value.